app/static/src/app/store/appState/actions.ts

```diff
@@ -29,5 +44,28 @@ export const appStateActions: ActionTree<AppState, AppState> = {
                 await dispatch(`model/${ModelAction.DefaultModel}`);
             }
         }
+    },
+
+    async [AppStateAction.QueueStateUpload](context) {
+        const { state, commit } = context;
+        const isFitting = () => { return (state.appType === AppType.Fit) && ((state as FitState).modelFit.fitting); };
+
+        // Do not queue uploads while fitting is true - we'll upload when fit finishes
+        if (!isFitting()) {
+            // remove any existing queued upload, as this request should supersede it
+            commit(AppStateMutation.ClearQueuedStateUpload);
+
+            const queuedId: number = window.setInterval(() => {
+                // wait for any ongoing uploads to finish before starting a new one
+                // and do not actually upload while fitting is true
+                if (!state.stateUploadInProgress && !isFitting()) {
```

Comment:
Because we don't want to queue when fitting, and also don't want to do the actual upload two seconds later if fitting then - and that value could theoretically have changed in the meantime (it's probably unlikely but could happen!).